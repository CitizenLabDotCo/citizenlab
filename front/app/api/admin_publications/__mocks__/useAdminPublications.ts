import { IAdminPublicationData } from '../types';

export const adminPublicationsData: IAdminPublicationData[] = [
  {
    id: 'b3b91b93-e207-4cb0-b49b-737d4b25792e',
    type: 'admin_publication',
    attributes: {
      ordering: 0,
      publication_status: 'published',
      depth: 0,
      publication_title_multiloc: {
        en: 'Quas excepturi ea et.',
        'fr-BE': 'Et illo et aperiam.',
        'nl-BE': 'Praesentium ut itaque ipsam.',
      },
      publication_description_multiloc: {
        en: '\u003cp\u003eAdipisci quia ducimus. Rem et ipsa. Ipsa vero quae.\u003c/p\u003e\u003cp\u003eVoluptas nemo sit. Tempora quia rerum. Ratione dolore quae.\u003c/p\u003e\u003cp\u003eNecessitatibus dolores quod. Magnam ullam natus. Culpa delectus rerum.\u003c/p\u003e',
        'fr-BE':
          '\u003cp\u003eLaudantium et et. Voluptates expedita ipsam. Omnis aut veritatis.\u003c/p\u003e\u003cp\u003eNon cupiditate quia. Illo laudantium tenetur. Neque aperiam aliquid.\u003c/p\u003e\u003cp\u003eAsperiores velit saepe. Veritatis quibusdam est. Fuga laudantium consequatur.\u003c/p\u003e',
        'nl-BE':
          '\u003cp\u003eDeserunt dolor quia. Quae ipsum omnis. Odit autem sint.\u003c/p\u003e\u003cp\u003eEt eos amet. Officia temporibus molestiae. Delectus nihil laboriosam.\u003c/p\u003e\u003cp\u003eMaiores eaque magni. Amet molestias asperiores. Incidunt magni consequatur.\u003c/p\u003e',
      },
      publication_description_preview_multiloc: {
        en: 'Et omnis recusandae minima.',
        'fr-BE': 'Blanditiis et repellendus cupiditate.',
        'nl-BE': 'Libero est voluptatibus tenetur.',
      },
      publication_slug: 'quas-excepturi-ea-et',
      visible_children_count: 1,
    },
    relationships: {
      publication: {
        data: { id: '99d51572-250c-48f4-8950-ea454c9476c1', type: 'folder' },
      },
      parent: { data: null },
      children: {
        data: [
          {
            id: '6012d402-3a56-47a8-8ce6-7cd1c69fa29d',
            type: 'admin_publication',
          },
        ],
      },
    },
  },
  {
    id: '8c8da82d-e4ea-4af4-8c7b-d221e86668cc',
    type: 'admin_publication',
    attributes: {
      ordering: 1,
      publication_status: 'archived',
      depth: 0,
      publication_title_multiloc: {
        en: 'Rem nulla consequatur vero.',
        'fr-BE': 'Ducimus voluptatem tempora expedita.',
        'nl-BE': 'Perferendis modi voluptatem facilis.',
      },
      publication_description_multiloc: {
        en: '\u003cp\u003eAd quam rerum. Fugiat vel dolores. Eligendi occaecati corporis.\u003c/p\u003e\u003cp\u003eDolorem est rem. Sequi aspernatur repudiandae. Tempora reprehenderit aut.\u003c/p\u003e\u003cp\u003eDolor accusantium molestiae. Pariatur aut tempora. Illum sequi illo.\u003c/p\u003e',
        'fr-BE':
          '\u003cp\u003eQuas rem sed. Temporibus corporis et. Fugiat et vero.\u003c/p\u003e\u003cp\u003eEsse aut dolorem. Quis impedit optio. Necessitatibus ipsa ut.\u003c/p\u003e\u003cp\u003eCorporis iure suscipit. Aut autem dolores. Voluptas ea eligendi.\u003c/p\u003e',
        'nl-BE':
          '\u003cp\u003eAliquam amet ut. Ea ut in. Architecto neque minus.\u003c/p\u003e\u003cp\u003eOmnis maxime ullam. Eaque eveniet illum. Animi aut accusantium.\u003c/p\u003e\u003cp\u003eQuasi consectetur dolorem. Ut quam assumenda. Aspernatur pariatur explicabo.\u003c/p\u003e',
      },
      publication_description_preview_multiloc: {
        en: 'Aspernatur ut eius excepturi.',
        'fr-BE': 'Esse incidunt dolore quos.',
        'nl-BE': 'Asperiores possimus labore sequi.',
      },
      publication_slug: 'rem-nulla-consequatur-vero',
      visible_children_count: 1,
    },
    relationships: {
      publication: {
        data: { id: 'c7f4f824-53e1-4b8c-866d-b3412f22d956', type: 'folder' },
      },
      parent: { data: null },
      children: {
        data: [
          {
            id: '8e3a6d01-994e-40d9-9657-9771a100adaf',
            type: 'admin_publication',
          },
        ],
      },
    },
  },
];

export const links = {
  self: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=1\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  first:
    'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=1\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  last: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=2\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  prev: null,
  next: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=2\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
};

export default jest.fn(() => {
  return { data: { pages: [{ data: adminPublicationsData, links }] } };
});
