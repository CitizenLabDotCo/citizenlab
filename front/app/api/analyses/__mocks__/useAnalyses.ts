import { IAnalysisData } from '../types';

export const analysesData: IAnalysisData[] = [
  {
    id: '1',
    type: 'analysis',
    attributes: {
      created_at: '2021-06-01T08:00:00.000Z',
      updated_at: '2021-06-01T08:00:00.000Z',
      participation_method: 'survey',
    },
    relationships: {
      project: {
        data: {
          id: '1',
          type: 'project',
        },
      },
      phase: {
        data: {
          id: '1',
          type: 'phase',
        },
      },
      custom_fields: {
        data: [
          {
            id: '1',
            type: 'custom_field',
          },
        ],
      },
    },
  },
];

export default jest.fn(() => {
  return { data: { data: analysesData } };
});
