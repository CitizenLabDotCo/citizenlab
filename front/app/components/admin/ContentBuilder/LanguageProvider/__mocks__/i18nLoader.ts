let translations: Record<string, Record<string, string>> = {
  en: { message: 'Hello' },
  'fr-FR': { message: 'Bonjour' },
};

export const i18nImports = new Proxy(
  {},
  {
    get(_, localePath: string) {
      const match = localePath.match(/\/i18n\/(.+)\.ts/);
      if (!match) {
        return undefined;
      }

      const locale = match[1];
      if (locale in translations) {
        return () =>
          Promise.resolve({
            default: translations[locale],
          });
      }

      return undefined;
    },
  }
);

export const setTranslations = (newTranslations: typeof translations) => {
  translations = newTranslations;
};
