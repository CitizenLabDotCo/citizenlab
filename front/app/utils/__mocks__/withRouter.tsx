import React from 'react';

export const withRouter = (Component) => {
  return (props) => {
    return <Component {...props} location={{ pathname: 'en' }} />;
  };
};
