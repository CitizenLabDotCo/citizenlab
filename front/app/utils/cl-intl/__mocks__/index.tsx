import React from 'react';
const Intl = jest.requireActual('react-intl');

// Initialise the real provider so that we don't
// need to reimplement any internals
const defaultProps = {
  locale: 'en',
  defaultLocale: 'en',
};
const intlProvider = new Intl.IntlProvider(defaultProps, {});

// The exact same `intl` object the real code receives ;-)
const { intl } = intlProvider.getChildContext();

const formatMessageReplacement = (
  messageDescriptor: ReactIntl.FormattedMessage.MessageDescriptor,
  values?: { [key: string]: string | number | boolean | Date } | undefined
) => {
  return intl.formatMessage(messageDescriptor, {
    tenantName: 'The Test',
    orgName: 'Test Town',
    orgType: 'testing',
    ...(values || {}),
  });
};

const intlReplacement = {
  ...intl,
  formatMessage: formatMessageReplacement,
};

const injectIntlReplacement = (Node) => {
  const renderWrapped = (props) => <Node {...props} intl={intl} />;
  renderWrapped.displayName = Node.displayName || Node.name || 'Component';
  return renderWrapped;
};

Intl.injectIntl = injectIntlReplacement;

// Override components by implementing the real components
//  providing them the context they need in order to function
const {
  IntlProvider,
  FormattedDate,
  FormattedTime,
  FormattedRelative,
  FormattedNumber,
  FormattedPlural,
  FormattedMessage,
  FormattedHTMLMessage,
} = Intl;

Intl.FormattedDate = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedDate {...props} />
  </IntlProvider>
);
Intl.FormattedTime = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedTime {...props} />
  </IntlProvider>
);
Intl.FormattedRelative = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedRelative {...props} />
  </IntlProvider>
);
Intl.FormattedNumber = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedNumber {...props} />
  </IntlProvider>
);
Intl.FormattedPlural = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedPlural {...props} />
  </IntlProvider>
);
Intl.FormattedMessage = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedMessage {...props} />
  </IntlProvider>
);
Intl.FormattedHTMLMessage = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedHTMLMessage {...props} />
  </IntlProvider>
);

// Set displayName so that snapshots don't use "Uknown" as component name
Intl.FormattedDate.displayName = 'FormattedDate';
Intl.FormattedTime.displayName = 'FormattedTime';
Intl.FormattedRelative.displayName = 'FormattedRelative';
Intl.FormattedNumber.displayName = 'FormattedNumber';
Intl.FormattedPlural.displayName = 'FormattedPlural';
Intl.FormattedMessage.displayName = 'FormattedMessage';
Intl.FormattedHTMLMessage.displayName = 'FormattedHTMLMessage';

// Special hook for tests, real package does not export this
Intl.intl = intlReplacement;

module.exports = Intl;
