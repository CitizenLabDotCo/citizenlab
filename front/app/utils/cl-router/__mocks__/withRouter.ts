import { WithRouterProps } from 'react-router';

export const mockWithRouterProps: WithRouterProps = {
  location: {
    pathname: 'some-page',
    search: '',
    query: {},
    state: '',
    hash: '',
    action: 'PUSH',
    key: '',
  },
  router: {
    push: () => null,
    replace: () => null,
    go: () => null,
    goBack: () => null,
    goForward: () => null,
    setRouteLeaveHook: () => null as any,
    createPath: () => '',
    createHref: () => '',
    isActive: () => true,
  },
  params: {},
  routes: [],
};
