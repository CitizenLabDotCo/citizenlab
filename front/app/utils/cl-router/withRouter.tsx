import React from 'react';
import {
  useLocation,
  // eslint-disable-next-line no-restricted-imports
  useNavigate,
  useParams,
  useSearchParams,
} from 'react-router-dom';

/** @deprecated Use `React Router hooks` instead */
export interface WithRouterProps {
  location: ReturnType<typeof useLocation> & { query: Record<string, any> };
  params: Record<string, string>;
  navigate: ReturnType<typeof useNavigate>;
}

/** @deprecated Use `React Router hooks` instead */
export const withRouter = <T extends WithRouterProps = WithRouterProps>(
  Component: React.ComponentType<T>
) => {
  return (props: Omit<T, keyof WithRouterProps>) => {
    const location = useLocation();
    const params = useParams();
    const navigate = useNavigate();
    const [searchParams] = useSearchParams();

    // normally you have to do searchParams.get('key'),
    // we create a normal key/value object here and pass it down
    const searchParamsObj = {};
    for (const queryParamsPair of searchParams.entries()) {
      const [key, val] = queryParamsPair;

      if (searchParamsObj[key] !== undefined) {
        if (Array.isArray(searchParamsObj[key])) {
          searchParamsObj[key].push(val);
        } else {
          const previousValue = searchParamsObj[key];
          searchParamsObj[key] = [previousValue, val];
        }
      } else {
        searchParamsObj[key] = val;
      }
    }

    return (
      <Component
        {...(props as T)}
        location={{ ...location, query: searchParamsObj }}
        params={params}
        navigate={navigate}
      />
    );
  };
};
