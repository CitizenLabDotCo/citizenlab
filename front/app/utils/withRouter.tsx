import React from 'react';
import {
  useLocation,
  useNavigate,
  useParams,
  useSearchParams,
} from 'react-router-dom';

export interface WithRouterProps {
  history: {
    back: () => void;
    goBack: () => void;
    location: ReturnType<typeof useLocation>;
    push: (url: string, state?: any) => void;
  };
  location: ReturnType<typeof useLocation> & { query: Record<string, string> };
  params: Record<string, string>;
  navigate: ReturnType<typeof useNavigate>;
}

export const withRouter = <P extends object>(Component: any) => {
  return (props: Omit<P, keyof WithRouterProps>) => {
    const location = useLocation();
    const params = useParams();
    const navigate = useNavigate();
    const [searchParams] = useSearchParams();

    const history = {
      back: () => navigate(-1),
      goBack: () => navigate(-1),
      location,
      push: (url: string, state?: any) => navigate(url, { state }),
      replace: (url: string, state?: any) =>
        navigate(url, {
          replace: true,
          state,
        }),
    };

    return (
      <Component
        history={history}
        location={{ ...location, query: searchParams }}
        params={params}
        navigate={navigate}
        {...(props as P)}
      />
    );
  };
};
