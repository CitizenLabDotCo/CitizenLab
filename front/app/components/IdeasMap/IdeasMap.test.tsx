import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { mockWithRouterProps } from '../../utils/cl-router/__mocks__/withRouter';
import { IdeasMap } from '.';
jest.mock('components/Outlet', () => 'Outlet');
jest.mock('modules', () => ({ streamsToReset: [] }));

jest.mock('components/Outlet', () => 'Outlet');

describe('IdeasMap', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <IdeasMap
        ideaMarkers={[]}
        windowSize={600}
        projectIds={['379cac45-d26b-505c-9215-b2677a693d37']}
        phaseId="c43968d3-171e-570c-8b71-f5fc0e55bbfa"
        {...mockWithRouterProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
