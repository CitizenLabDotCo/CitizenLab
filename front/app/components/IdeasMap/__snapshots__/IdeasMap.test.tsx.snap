// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeasMap renders correctly 1`] = `
<IdeasMap__Container>
  <a11y__ScreenReaderOnly>
    <FormattedMessage
      defaultMessage="Map overview"
      id="app.components.IdeasMap.a11y_mapTitle"
    />
  </a11y__ScreenReaderOnly>
  <Memo()
    boxContent={null}
    fitBounds={true}
    onBoxClose={[Function]}
    onMapClick={[Function]}
    onMarkerClick={[Function]}
    points={Array []}
    projectId="379cac45-d26b-505c-9215-b2677a693d37"
  />
  <IdeasMap__IdeaButtonWrapper
    className="create-idea-wrapper"
  >
    <_default
      inMap={true}
      latLng={null}
      participationContextType="phase"
      phaseId="c43968d3-171e-570c-8b71-f5fc0e55bbfa"
      projectId="379cac45-d26b-505c-9215-b2677a693d37"
    />
  </IdeasMap__IdeaButtonWrapper>
</IdeasMap__Container>
`;
